analyses/DESeq_analysis/run-DESeq-analysis.R

```diff
@@ -0,0 +1,200 @@
+# Author: Sangeeta Shukla and Alvin Farrel
+# Date: June 2021
+# Function: 
+# 1. summarize Differential expression from RNASeq data
+# 2. tabulate corresponding P-value
+
+# Example run: DESeq
+# Rscript analyses/DESeq/run-DESeq-analysis.R 1 1
+# This was compares 1 cancer group (Combined or Cohort specific) with one GTEx tissue type. The input arguments are the indices of the comparison of the cancer groups and gtex sub tissues to be compared. In v6, there are 107 cancer groups vs 54 GTEx tissues
+
+# Read arguments from terminal
+args <- commandArgs(trailingOnly = TRUE)
+
+# Load required libraries
+suppressPackageStartupMessages({
+  library(ggplot2)
+  library(DESeq2)
+})
+
+
+HIST_index <- args[1]   #assign first argument to Histology index
+GTEX_index <- args[2]   #assign second argument to GTEx index
+
+    
+
+
+
+#Load histology file
+hist <- read.delim("../data/v6/histologies.tsv", header=TRUE, sep = '\t')
+
+#Load expression counts data
+countData <- readRDS("../data/v6/gene-counts-rsem-expected_count-collapsed.rds")
+
+#Load expression TPM data
+TPMData <- readRDS("../data/v6/gene-expression-rsem-tpm-collapsed.rds")
+
+#Load EFO-MONDO map file
+EFO_MONDO <- read.delim("../data/v6/efo-mondo-map.tsv", header =T, stringsAsFactors = FALSE)
+
+#Load gene symbol-gene ID RMTL file
+ENSG_Hugo <- read.delim("../data/v6/ensg-hugo-rmtl-v1-mapping.tsv", header =T)
+
+# Subset Histology file for samples only found in the current the countData file (To ensure no discepancies cause errors later in the code)
+hist.filtered <- unique(hist[which(hist$Kids_First_Biospecimen_ID %in%  colnames(countData)),])
+
+#Subset countdata for data that are present in the hitstoly files (To ensure no discepancies cause errors later in the code)
+countData_filtered <- countData[,which(colnames(countData) %in% hist$Kids_First_Biospecimen_ID)]
+
+#Subset countadata for data that are present in the hitstoly files (To ensure no discepancies cause errors later in the code)
+TPMData_filtered <- TPMData[,which(colnames(TPMData) %in% hist$Kids_First_Biospecimen_ID)]
+
```

Comment:
Hi @jharenza . I was wondering if other modules should also use all-cohort independent sample list for all-cohort analysis and use each-cohort independent sample list for each-cohort analysis. Sorry for the digression from this PR.

My concern is that all-cohort and each-cohort independent sample lists currently have very different independent samples due to the random selection procedures, which may cause result discrepancies between only-one-cohort-all-cohort and each-cohort. Such potential discrepancies could be reduced by updating the independent-samples module to favor selecting the same biospecimen IDs that are in the all-cohort sample lists. I estimate this refactoring may take a week, including review time. Even after this refactoring, there will still be result discrepancies between only-one-cohort-all-cohort and each-cohort, but these discrepancies will not be caused by random selection but rather overlapping patients between cohorts.

Reply:
>I was wondering if other modules should also use all-cohort independent sample list for all-cohort analysis and use each-cohort independent sample list for each-cohort analysis. 

Yes, this is the plan so that all analyses for RNA and DNA use the same lists, which may be updated per release if/when new data is added.

>My concern is that all-cohort and each-cohort independent sample lists currently have very different independent samples due to the random selection procedures

This is a good point and makes sense to do - can you submit a ticket for that? I would perhaps call that a medium priority task.

Reply:
I agree that all analyses should use the same lists.

Sure. I will submit a ticket for updating the independent sample lists. I will also submit a couple of other tickets for analysis modules to update the independent sample lists being used and label them with blocked and referring to the independent sample lists update ticket, so that all-cohort analysis uses all-cohort independent lists, and each-cohort analysis uses each-cohort independent sample lists.

Reply:
Thanks!